sql/src/main/java/io/crate/analyze/MultiSourceSelect.java

```diff
@@ -69,6 +68,10 @@ public MultiSourceSelect(
         return sources;
     }
 
+    public List<JoinPair> joinPairs() {
+        return joinPairs;
```

Comment:
it must be mutable, otherwise we'd have to copy it to a mutable list. see https://github.com/crate/crate/pull/4005/files#diff-9f693a5327e3ead9f38749c2504a673aR44